lib/firenest/replicated_state.ex

```diff
@@ -0,0 +1,343 @@
+defmodule Firenest.ReplicatedState do
+  @moduledoc """
+  Distributed key-value store for ephemeral data.
+
+  The key-value pairs are always attached to a lifetime of a
+  process and the state is replicated to all nodes across the
+  topology. When the attached process dies the state will be
+  removed on all nodes and in case the all the state from
+  disconnected nodes will be (temporarily) removed. The state
+  can only be attached to local processes.
```

Comment:
Just a note: I remember in our initial designs we also discussed the option of having ephemeral processes by means of timeouts. They are not attached to any processes but are guaranteed to eventually expire.

Truth be told, we don't have to attach them to processes, the only meaning we need is that **the data is lost once the node it belongs to is gone**.

Reply:
Oh, I even see that this behaviour is implemented as the `:partition` key. Beautiful.